ktor-client/ktor-client-core/src/io/ktor/client/request/HttpRequest.kt

```diff
@@ -88,7 +88,7 @@ class HttpRequestBuilder : HttpMessageBuilder {
      */
     val executionContext: CompletableDeferred<Unit> = CompletableDeferred()
 
-    private var attributesBuilder: Attributes.() -> Unit = {}
+    var attributesBuilder: Attributes.() -> Unit = {}
```

Comment:
It would be better to have `takeFrom` method inside the class so we do not expose `attributesBuilder`. What to you think?

Reply:
Yep. Let's add the method.